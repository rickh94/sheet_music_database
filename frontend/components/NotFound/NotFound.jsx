import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Header from '../Header'

import Container from 'react-bulma-components/lib/components/container'
import Card from 'react-bulma-components/lib/components/card/card'
import Heading from 'react-bulma-components/lib/components/heading'

class NotFound extends Component {
  render() {
    return (
      <React.Fragment>
        <Header />
        <Container>
          <Card className="margin-default">
            <Card.Header>
              <Card.Header.Title>
                <Heading size={5} className='no-margin-padding'>Not Found</Heading>
              </Card.Header.Title>
            </Card.Header>
            <Card.Content>The page you are looking for could not be found</Card.Content>
          </Card>
        </Container>
      </React.Fragment>
    )
  }
}

NotFound.propTypes = {}

export default NotFound
