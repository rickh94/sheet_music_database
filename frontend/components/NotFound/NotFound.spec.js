import React from 'react'
import { shallow } from 'enzyme'
import NotFound from './NotFound'

import '../../setupTests'

describe('<NotFound />', () => {
  const wrapper = shallow(<NotFound />)
  beforeEach(() => {
    wrapper.update()
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the Header', () => {
    expect(wrapper.exists('Header')).toBeTruthy()
  })

  it('renders a container', () => {
    expect(wrapper.exists('Container')).toBeTruthy()
  })

  it('renders a Card with some margin', () => {
    const card = wrapper.find('Card')
    expect(card.length).toEqual(1)
    expect(card.hasClass('margin-default')).toBeTruthy()
  })
})
