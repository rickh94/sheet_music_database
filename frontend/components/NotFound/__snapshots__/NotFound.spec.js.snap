// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <Container
          breakpoint={null}
          fluid={false}
          renderAs="div"
          style={Object {}}
        >
          <Card
            className="margin-default"
            renderAs="div"
            style={Object {}}
          >
            <CardHeader
              renderAs="div"
            >
              <CardHeaderTitle
                renderAs="div"
              >
                <Heading
                  className="no-margin-padding"
                  heading={false}
                  renderAs="h1"
                  size={5}
                  spaced={false}
                  subtitle={false}
                  weight={null}
                >
                  Not Found
                </Heading>
              </CardHeaderTitle>
            </CardHeader>
            <CardContent
              renderAs="div"
            >
              The page you are looking for could not be found
            </CardContent>
          </Card>
        </Container>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "backgroundColor": undefined,
          "breakpoint": null,
          "children": <Card
            className="margin-default"
            renderAs="div"
            style={Object {}}
          >
            <CardHeader
              renderAs="div"
            >
              <CardHeaderTitle
                renderAs="div"
              >
                <Heading
                  className="no-margin-padding"
                  heading={false}
                  renderAs="h1"
                  size={5}
                  spaced={false}
                  subtitle={false}
                  weight={null}
                >
                  Not Found
                </Heading>
              </CardHeaderTitle>
            </CardHeader>
            <CardContent
              renderAs="div"
            >
              The page you are looking for could not be found
            </CardContent>
          </Card>,
          "className": undefined,
          "clearfix": undefined,
          "clipped": undefined,
          "fluid": false,
          "hidden": undefined,
          "invisible": undefined,
          "italic": undefined,
          "marginless": undefined,
          "overlay": undefined,
          "paddingless": undefined,
          "pull": undefined,
          "radiusless": undefined,
          "renderAs": "div",
          "responsive": undefined,
          "shadowless": undefined,
          "style": Object {},
          "textAlignment": undefined,
          "textColor": undefined,
          "textSize": undefined,
          "textTransform": undefined,
          "textWeight": undefined,
          "unselectable": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backgroundColor": undefined,
            "children": Array [
              <CardHeader
                renderAs="div"
              >
                <CardHeaderTitle
                  renderAs="div"
                >
                  <Heading
                    className="no-margin-padding"
                    heading={false}
                    renderAs="h1"
                    size={5}
                    spaced={false}
                    subtitle={false}
                    weight={null}
                  >
                    Not Found
                  </Heading>
                </CardHeaderTitle>
              </CardHeader>,
              <CardContent
                renderAs="div"
              >
                The page you are looking for could not be found
              </CardContent>,
            ],
            "className": "margin-default",
            "clearfix": undefined,
            "clipped": undefined,
            "hidden": undefined,
            "invisible": undefined,
            "italic": undefined,
            "marginless": undefined,
            "overlay": undefined,
            "paddingless": undefined,
            "pull": undefined,
            "radiusless": undefined,
            "renderAs": "div",
            "responsive": undefined,
            "shadowless": undefined,
            "style": Object {},
            "textAlignment": undefined,
            "textColor": undefined,
            "textSize": undefined,
            "textTransform": undefined,
            "textWeight": undefined,
            "unselectable": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "backgroundColor": undefined,
                "children": <CardHeaderTitle
                  renderAs="div"
                >
                  <Heading
                    className="no-margin-padding"
                    heading={false}
                    renderAs="h1"
                    size={5}
                    spaced={false}
                    subtitle={false}
                    weight={null}
                  >
                    Not Found
                  </Heading>
                </CardHeaderTitle>,
                "className": undefined,
                "clearfix": undefined,
                "clipped": undefined,
                "hidden": undefined,
                "invisible": undefined,
                "italic": undefined,
                "marginless": undefined,
                "overlay": undefined,
                "paddingless": undefined,
                "pull": undefined,
                "radiusless": undefined,
                "renderAs": "div",
                "responsive": undefined,
                "shadowless": undefined,
                "textAlignment": undefined,
                "textColor": undefined,
                "textSize": undefined,
                "textTransform": undefined,
                "textWeight": undefined,
                "unselectable": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "backgroundColor": undefined,
                  "children": <Heading
                    className="no-margin-padding"
                    heading={false}
                    renderAs="h1"
                    size={5}
                    spaced={false}
                    subtitle={false}
                    weight={null}
                  >
                    Not Found
                  </Heading>,
                  "className": undefined,
                  "clearfix": undefined,
                  "clipped": undefined,
                  "hidden": undefined,
                  "invisible": undefined,
                  "italic": undefined,
                  "marginless": undefined,
                  "overlay": undefined,
                  "paddingless": undefined,
                  "pull": undefined,
                  "radiusless": undefined,
                  "renderAs": "div",
                  "responsive": undefined,
                  "shadowless": undefined,
                  "textAlignment": undefined,
                  "textColor": undefined,
                  "textSize": undefined,
                  "textTransform": undefined,
                  "textWeight": undefined,
                  "unselectable": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "backgroundColor": undefined,
                    "children": "Not Found",
                    "className": "no-margin-padding",
                    "clearfix": undefined,
                    "clipped": undefined,
                    "heading": false,
                    "hidden": undefined,
                    "invisible": undefined,
                    "italic": undefined,
                    "marginless": undefined,
                    "overlay": undefined,
                    "paddingless": undefined,
                    "pull": undefined,
                    "radiusless": undefined,
                    "renderAs": "h1",
                    "responsive": undefined,
                    "shadowless": undefined,
                    "size": 5,
                    "spaced": false,
                    "subtitle": false,
                    "textAlignment": undefined,
                    "textColor": undefined,
                    "textSize": undefined,
                    "textTransform": undefined,
                    "textWeight": undefined,
                    "unselectable": undefined,
                    "weight": null,
                  },
                  "ref": null,
                  "rendered": "Not Found",
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "backgroundColor": undefined,
                "children": "The page you are looking for could not be found",
                "className": undefined,
                "clearfix": undefined,
                "clipped": undefined,
                "hidden": undefined,
                "invisible": undefined,
                "italic": undefined,
                "marginless": undefined,
                "overlay": undefined,
                "paddingless": undefined,
                "pull": undefined,
                "radiusless": undefined,
                "renderAs": "div",
                "responsive": undefined,
                "shadowless": undefined,
                "textAlignment": undefined,
                "textColor": undefined,
                "textSize": undefined,
                "textTransform": undefined,
                "textWeight": undefined,
                "unselectable": undefined,
              },
              "ref": null,
              "rendered": "The page you are looking for could not be found",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <Container
            breakpoint={null}
            fluid={false}
            renderAs="div"
            style={Object {}}
          >
            <Card
              className="margin-default"
              renderAs="div"
              style={Object {}}
            >
              <CardHeader
                renderAs="div"
              >
                <CardHeaderTitle
                  renderAs="div"
                >
                  <Heading
                    className="no-margin-padding"
                    heading={false}
                    renderAs="h1"
                    size={5}
                    spaced={false}
                    subtitle={false}
                    weight={null}
                  >
                    Not Found
                  </Heading>
                </CardHeaderTitle>
              </CardHeader>
              <CardContent
                renderAs="div"
              >
                The page you are looking for could not be found
              </CardContent>
            </Card>
          </Container>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backgroundColor": undefined,
            "breakpoint": null,
            "children": <Card
              className="margin-default"
              renderAs="div"
              style={Object {}}
            >
              <CardHeader
                renderAs="div"
              >
                <CardHeaderTitle
                  renderAs="div"
                >
                  <Heading
                    className="no-margin-padding"
                    heading={false}
                    renderAs="h1"
                    size={5}
                    spaced={false}
                    subtitle={false}
                    weight={null}
                  >
                    Not Found
                  </Heading>
                </CardHeaderTitle>
              </CardHeader>
              <CardContent
                renderAs="div"
              >
                The page you are looking for could not be found
              </CardContent>
            </Card>,
            "className": undefined,
            "clearfix": undefined,
            "clipped": undefined,
            "fluid": false,
            "hidden": undefined,
            "invisible": undefined,
            "italic": undefined,
            "marginless": undefined,
            "overlay": undefined,
            "paddingless": undefined,
            "pull": undefined,
            "radiusless": undefined,
            "renderAs": "div",
            "responsive": undefined,
            "shadowless": undefined,
            "style": Object {},
            "textAlignment": undefined,
            "textColor": undefined,
            "textSize": undefined,
            "textTransform": undefined,
            "textWeight": undefined,
            "unselectable": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "backgroundColor": undefined,
              "children": Array [
                <CardHeader
                  renderAs="div"
                >
                  <CardHeaderTitle
                    renderAs="div"
                  >
                    <Heading
                      className="no-margin-padding"
                      heading={false}
                      renderAs="h1"
                      size={5}
                      spaced={false}
                      subtitle={false}
                      weight={null}
                    >
                      Not Found
                    </Heading>
                  </CardHeaderTitle>
                </CardHeader>,
                <CardContent
                  renderAs="div"
                >
                  The page you are looking for could not be found
                </CardContent>,
              ],
              "className": "margin-default",
              "clearfix": undefined,
              "clipped": undefined,
              "hidden": undefined,
              "invisible": undefined,
              "italic": undefined,
              "marginless": undefined,
              "overlay": undefined,
              "paddingless": undefined,
              "pull": undefined,
              "radiusless": undefined,
              "renderAs": "div",
              "responsive": undefined,
              "shadowless": undefined,
              "style": Object {},
              "textAlignment": undefined,
              "textColor": undefined,
              "textSize": undefined,
              "textTransform": undefined,
              "textWeight": undefined,
              "unselectable": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "backgroundColor": undefined,
                  "children": <CardHeaderTitle
                    renderAs="div"
                  >
                    <Heading
                      className="no-margin-padding"
                      heading={false}
                      renderAs="h1"
                      size={5}
                      spaced={false}
                      subtitle={false}
                      weight={null}
                    >
                      Not Found
                    </Heading>
                  </CardHeaderTitle>,
                  "className": undefined,
                  "clearfix": undefined,
                  "clipped": undefined,
                  "hidden": undefined,
                  "invisible": undefined,
                  "italic": undefined,
                  "marginless": undefined,
                  "overlay": undefined,
                  "paddingless": undefined,
                  "pull": undefined,
                  "radiusless": undefined,
                  "renderAs": "div",
                  "responsive": undefined,
                  "shadowless": undefined,
                  "textAlignment": undefined,
                  "textColor": undefined,
                  "textSize": undefined,
                  "textTransform": undefined,
                  "textWeight": undefined,
                  "unselectable": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "backgroundColor": undefined,
                    "children": <Heading
                      className="no-margin-padding"
                      heading={false}
                      renderAs="h1"
                      size={5}
                      spaced={false}
                      subtitle={false}
                      weight={null}
                    >
                      Not Found
                    </Heading>,
                    "className": undefined,
                    "clearfix": undefined,
                    "clipped": undefined,
                    "hidden": undefined,
                    "invisible": undefined,
                    "italic": undefined,
                    "marginless": undefined,
                    "overlay": undefined,
                    "paddingless": undefined,
                    "pull": undefined,
                    "radiusless": undefined,
                    "renderAs": "div",
                    "responsive": undefined,
                    "shadowless": undefined,
                    "textAlignment": undefined,
                    "textColor": undefined,
                    "textSize": undefined,
                    "textTransform": undefined,
                    "textWeight": undefined,
                    "unselectable": undefined,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "backgroundColor": undefined,
                      "children": "Not Found",
                      "className": "no-margin-padding",
                      "clearfix": undefined,
                      "clipped": undefined,
                      "heading": false,
                      "hidden": undefined,
                      "invisible": undefined,
                      "italic": undefined,
                      "marginless": undefined,
                      "overlay": undefined,
                      "paddingless": undefined,
                      "pull": undefined,
                      "radiusless": undefined,
                      "renderAs": "h1",
                      "responsive": undefined,
                      "shadowless": undefined,
                      "size": 5,
                      "spaced": false,
                      "subtitle": false,
                      "textAlignment": undefined,
                      "textColor": undefined,
                      "textSize": undefined,
                      "textTransform": undefined,
                      "textWeight": undefined,
                      "unselectable": undefined,
                      "weight": null,
                    },
                    "ref": null,
                    "rendered": "Not Found",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "backgroundColor": undefined,
                  "children": "The page you are looking for could not be found",
                  "className": undefined,
                  "clearfix": undefined,
                  "clipped": undefined,
                  "hidden": undefined,
                  "invisible": undefined,
                  "italic": undefined,
                  "marginless": undefined,
                  "overlay": undefined,
                  "paddingless": undefined,
                  "pull": undefined,
                  "radiusless": undefined,
                  "renderAs": "div",
                  "responsive": undefined,
                  "shadowless": undefined,
                  "textAlignment": undefined,
                  "textColor": undefined,
                  "textSize": undefined,
                  "textTransform": undefined,
                  "textWeight": undefined,
                  "unselectable": undefined,
                },
                "ref": null,
                "rendered": "The page you are looking for could not be found",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
